commit 60eb5da2434b5288291aac533fb4dadc8ec8aed7
Author: Arnd Bergmann <arnd@arndb.de>
Date:   Fri Nov 2 12:11:22 2018 +0100

    ubifs: auth: Add CONFIG_KEYS dependency
    
    The new authentication support causes a build failure
    when CONFIG_KEYS is disabled, so add a dependency.
    
    fs/ubifs/auth.c: In function 'ubifs_init_authentication':
    fs/ubifs/auth.c:249:16: error: implicit declaration of function 'request_key'; did you mean 'request_irq'? [-Werror=implicit-function-declaration]
      keyring_key = request_key(&key_type_logon, c->auth_key_name, NULL);
    
    Fixes: d8a22773a12c ("ubifs: Enable authentication support")
    Signed-off-by: Arnd Bergmann <arnd@arndb.de>
    Signed-off-by: Richard Weinberger <richard@nod.at>

diff --git a/fs/ubifs/Kconfig b/fs/ubifs/Kconfig
index 3ee613361f4e..bc1e082d921d 100644
--- a/fs/ubifs/Kconfig
+++ b/fs/ubifs/Kconfig
@@ -86,6 +86,7 @@ config UBIFS_FS_SECURITY
 
 config UBIFS_FS_AUTHENTICATION
 	bool "UBIFS authentication support"
+	depends on KEYS
 	select CRYPTO_HMAC
 	help
 	  Enable authentication support for UBIFS. This feature offers protection